### PR TITLE
[JENKINS-55703] - Add a  “—runWorkspace” parameter to define the workspace 

### PR DESCRIPTION
**Pull request by oleg-nenashev**
Also, the new parameter is enabled by default for vanilla Docker images.

### CODE REVIEW

**Review by fcojfernandez — APPROVED** [review]

**Comment by oleg-nenashev** [comment]
So the tricky thing here is to get it integrated with Custom WAR Packager. It would require bumping  minimum JFR requirement there (or to pass JAVA_OPTS) there instead

File: bootstrap/src/main/java/io/jenkins/jenkinsfile/runner/bootstrap/Bootstrap.java [hunk 1]
```diff
@@ -5,6 +5,7 @@
 import org.kohsuke.args4j.CmdLineParser;
 import org.kohsuke.args4j.Option;
 
+import javax.annotation.CheckForNull;
```
Comment (varyvol) [inline bootstrap/src/main/java/io/jenkins/jenkinsfile/runner/bootstrap/Bootstrap.java:8]: You added FIndbugs but the annotation is the one from javax?

File: bootstrap/src/main/java/io/jenkins/jenkinsfile/runner/bootstrap/Bootstrap.java [hunk 2]
```diff
@@ -109,6 +119,15 @@ private void postConstruct() throws IOException {
                 installPlugin(shortname, version);
             }
         }
+
+        if (this.runWorkspace != null){
+            if (System.getProperty("jenkins.model.Jenkins.workspacesDir") != null) {
+                //TODO(oleg_nenashev): It would have been better to keep it other way, but made it as is to retain compatibility
+                System.out.println("Ignoring the --runWorkspace argument, because an explicit System property is set");
```
Comment (varyvol) [inline bootstrap/src/main/java/io/jenkins/jenkinsfile/runner/bootstrap/Bootstrap.java]: Maybe we can provide the name of the system property in the message?

File: bootstrap/src/main/java/io/jenkins/jenkinsfile/runner/bootstrap/Bootstrap.java [hunk 3]
```diff
@@ -109,6 +119,15 @@ private void postConstruct() throws IOException {
                 installPlugin(shortname, version);
             }
         }
+
+        if (this.runWorkspace != null){
+            if (System.getProperty("jenkins.model.Jenkins.workspacesDir") != null) {
```
Comment (jvz) [inline bootstrap/src/main/java/io/jenkins/jenkinsfile/runner/bootstrap/Bootstrap.java]: You have this property used multiple times. Want to refactor into a constant?

Comment (oleg-nenashev) [reply bootstrap/src/main/java/io/jenkins/jenkinsfile/runner/bootstrap/Bootstrap.java]: no. but i will

File: bootstrap/src/main/java/io/jenkins/jenkinsfile/runner/bootstrap/Bootstrap.java (hunk 1 above)
Comment (oleg-nenashev) [reply bootstrap/src/main/java/io/jenkins/jenkinsfile/runner/bootstrap/Bootstrap.java:8]: Yes. It comes from `com.google.code.findbugs`

**Review by varyvol — APPROVED** [review]

**Review by fcojfernandez — APPROVED** [review]
Re-approving

**Pull request merged**